pkgs/os-specific/linux/linux-laptop-system-tray-extensions/state.patch

```diff
@@ -0,0 +1,11 @@
+--- a/modules/Core/main.py     2021-12-14 23:16:06.182256726 +0100
++++ b/modules/Core/main.py     2021-12-15 22:43:37.415008227 +0100
+@@ -178,7 +178,7 @@
+         self.exit_code = 0
+ 
+         # init state manager
+-        self.state_manager = StateManager(filename=self.project_path + "/state.json")
++        self.state_manager = StateManager(filename=os.path.expanduser("~") + "/.local/ste/state.json")
```

Comment:
Maybe make this more properly XDG?

```suggestion
        self.state_manager = StateManager(filename=os.getenv("XDG_STATE_HOME", os.path.expanduser("~") + "/.local/state") + "/ste/state.json")
```

Reply:
I tried this, and it still works (`XDG_STATE_HOME` isn't set over here, but your suggestion falls back to `os.path.expanduser("~")`). What's the purpose of `XDG_STATE_HOME`? It sounds like it's made specifically for referring to this state folder. I made the default path `/.local/ste/state.json` because otherwise STE would write the user-settings to the project root (which is read-only in NixOS as it's under `/nix/store`).

If we want users to choose their own location for storing this program's settings, wouldn't it be better to make a less "generic" named env var like `STE_CONFIG_PATH`? It sounds like `XDG_STATE_HOME` is something we can easily conflict with if another package or application uses this var.

Reply:
`XDG_STATE_HOME` is expected to be used by various packages. See https://specifications.freedesktop.org/basedir-spec/latest/ar01s03.html for the xdg spec with respect to env vars.

Using `~/.local/ste` doesn't fit with XDG. XDG doesn't have applications place files/dirs directly into `.local`, but instead wants them to use subdirs in `.local` (like `state`) depending on the use case of files/dirs.

Reply:
> `XDG_STATE_HOME` is expected to be used by various packages. See https://specifications.freedesktop.org/basedir-spec/latest/ar01s03.html for the xdg spec with respect to env vars.
> 
> Using `~/.local/ste` doesn't fit with XDG. XDG doesn't have applications place files/dirs directly into `.local`, but instead wants them to use subdirs in `.local` (like `state`) depending on the use case of files/dirs.

Thanks a lot for clarifying, the "_STATE_" part confused as I thought it was named after the "state" folder in this package.

Reply:
I think my solution is quite simple and will work for most people. Those who desire more customisation can just set `XDG_STATE_HOME` for this package specifically, or simply patch it.

Reply:
Alright I'll change it like this, if everyone is ok with it. Thanks very much!